rust/foxglove-derive/src/lib.rs

```diff
@@ -68,6 +76,10 @@ fn derive_enum_impl(input: &DeriveInput, data: &DataEnum) -> TokenStream {
             fn type_name() -> Option<String> {
                 Some(stringify!(#name).to_string())
             }
+
+            fn encoded_len(&self) -> usize {
+                ::foxglove::protobuf::encoded_len_varint(*self as u64)
```

Comment:
Why does this cast self to u64?

Reply:
The cast came from the original implementation of `write` in the base branch, which was basically doing the same thing as prost, and prost's interface accepts u64. It works for the standard varint encoding because both [protobuf](https://protobuf.dev/programming-guides/encoding/#signed-ints) and [rust](https://doc.rust-lang.org/book/ch03-02-data-types.html#integer-types) use two's complement for signed ints. Should we do something more explicit here?